server/schedulers/balance_plan.go

```diff
@@ -80,3 +98,66 @@ func (p *balanceSchedulerPlan) Clone(opts ...plan.Option) plan.Plan {
 	}
 	return plan
 }
+
+func BalancePlanSummary(plans []plan.Plan) (string, error) {
+	// storeStatusCounter is used to count the number of various statuses of each store
+	var storeStatusCounter map[uint64]map[plan.Status]int
+	// statusCounter is used to count the number of status which is regarded as best status of each store
+	statusCounter := make(map[plan.Status]uint64)
+	maxStep := -1
+	for _, pi := range plans {
+		p, ok := pi.(*balanceSchedulerPlan)
+		if !ok {
+			return "", errs.ErrDiagnoseLoadPlanError
+		}
+		step := p.GetStep()
+		// we don't consider the situation for verification step
+		if step > 2 {
+			continue
+		}
+		if step > maxStep {
+			storeStatusCounter = make(map[uint64]map[plan.Status]int)
+			maxStep = p.GetStep()
+		} else if step < maxStep {
+			continue
+		}
+		var store uint64
+		if step == 1 {
+			store = p.source.GetID()
+		} else {
+			store = p.GetResource(p.GetStep())
```

Comment:
Why not use `p.GetResource(p.GetStep())` direcly?

Reply:
This is a special processing in summary, because I want to exclude the factor of region and consider the failure  as the status of store.